src/Checkout.js

```diff
@@ -50,13 +52,26 @@ export default function Checkout() {
     canSimulate: false,
     comments: ''
   });
+  const [idToken, setIdToken] = useState('');
 
   useEffect(() => {
-    fetchMentorData();
+      firebase.auth().onAuthStateChanged(user => {
+        if (user) {
+          user.getIdToken().then(token => {
+            setIdToken(token);
+          })
+        }
+      });
+
   }, []);
 
-  async function fetchMentorData() {
-    const mentorData = await fetchMentor(tempMentorId);
+  useEffect(() => {
+    if (idToken !== '') {
+      fetchMentorData(idToken);
+    }}, [idToken]);
+
```

Comment:
getIdToken uses cache, and I don't want people to consider saving the result somewhere in our code (I want just to have the SDK do that). In addition, you don't need to configure a new listener, there's a currentuser object on firebase.auth() that you can get the token from, if I'm not mistaken. So please make this change: call getIdToken not in this file but in the API file (where updateMentor is from, from within the axios call), for each api. We can then do a performance review/improvement round.